src/main/java/seedu/address/model/person/RoleContainsKeywordsPredicate.java

```diff
@@ -1,32 +1,55 @@
 package seedu.address.model.person;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Predicate;
 
+import seedu.address.commons.util.ToStringBuilder;
+
 /**
- * Represents a predicate that checks if a person's role contains a specified keyword.
+ * Represents a predicate that checks if a person's role contains any of the specified keywords.
  * This predicate is used to filter a list of persons based on their roles.
  */
 public class RoleContainsKeywordsPredicate implements Predicate<Person> {
-    private final String keyword;
+    private final List<String> keywords;
 
     /**
-     * Constructs a {@code RoleContainsKeywordsPredicate} with the specified keyword.
+     * Constructs a {@code RoleContainsKeywordsPredicate} with a single keyword.
      *
      * @param keyword The keyword to match against the person's role.
      */
     public RoleContainsKeywordsPredicate(String keyword) {
-        this.keyword = keyword.toLowerCase();
+        this.keywords = new ArrayList<>();
+        this.keywords.add(keyword.toLowerCase());
+    }
+
+    /**
+     * Constructs a {@code RoleContainsKeywordsPredicate} with a list of keywords.
+     *
+     * @param keywords The list of keywords to match against the person's role.
+     */
+    public RoleContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = new ArrayList<>();
+        for (String keyword : keywords) {
+            this.keywords.add(keyword.toLowerCase());
+        }
     }
 
     /**
-     * Tests if the given person's role contains the keyword.
+     * Tests if the given person's role contains any of the keywords.
      *
      * @param person The person to test.
-     * @return {@code true} if the person's role contains the keyword, {@code false} otherwise.
+     * @return {@code true} if the person's role contains any of the keywords, {@code false} otherwise.
      */
     @Override
     public boolean test(Person person) {
-        return person.getRole().toLowerCase().equals(keyword);
+        String lowercaseRole = person.getRole().toLowerCase();
+        for (String keyword : keywords) {
+            if (lowercaseRole.contains(keyword)) {
```

Comment:
Any reason we use contains when FindRoleCommandParser require an exact input of either `client`, `employee`, `supplier`?